diff --git a/backends/sqlite3/standard-use-type.cpp b/backends/sqlite3/standard-use-type.cpp
index 43ed4f0..123c075 100644
--- a/backends/sqlite3/standard-use-type.cpp
+++ b/backends/sqlite3/standard-use-type.cpp
@@ -74,7 +74,9 @@ void sqlite3_standard_use_type_backend::pre_use(indicator const * ind)
 
     if (statement_.useData_[0].size() < static_cast<std::size_t>(position_))
     {
-        statement_.useData_[0].resize(position_);
+        sqlite3_column empty;
+        empty.isNull_ = true;
+        statement_.useData_[0].resize(position_, empty);
     }
 
     if (ind != NULL && *ind == i_null)
